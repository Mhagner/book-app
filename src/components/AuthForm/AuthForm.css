.auth-form__container {
    padding: 2rem;
}

.auth-form__box-field {
    position: relative;
    padding: 0.5rem 1.6rem;
    width: 32rem;
    background-color: rgba(0, 0, 0, 0.32);
    height: 6rem;
    display: flex;
    flex-direction: column;
    border-radius: 4px;
    margin-bottom: 1.6rem;
    transition: all .2s;
}

.auth-form__box-field:hover {
    border: 1px solid var(--color-white);
}

.auth-form__field {
    font-size: 1.6rem;
    background-color: transparent;
    border: none;
    outline: none;
    height: 24px;
    color: var(--color-white);
}

.auth-form__box-field label {
    font-size: 1.2rem;
    color: var(--color-white);
    opacity: 0.5;
}

.auth-form__button {
    position: absolute;
    width: 8.5rem;
    height: 3.6rem;
    padding: 0.8rem;
    border-radius: 44px;
    border: none;
    top: 12px;
    right: 16px;
    font-size: 1.6rem;
    color: var(--color-primary-light);
    font-weight: bold;
    transition: all 0.2s
}

.auth-form__button:hover {
    transform: scale(1.1);
}

.error-backend ,
.auth-form__error {
    background: rgba(255, 255, 255, 0.4);
    color: var(--color-white);
    position: absolute;
    border-radius: 0.4rem;
}

.auth-form__error {
    padding: 0.4rem;
    right: 1.2rem;
    top: 5rem;
    font-size: 1.2rem;
}

.error-backend {
    padding: 1.4rem;
    top: 8rem;
    left: 0;
    font-size: 1.6rem;
}

@media (min-width: 1026px) {
    .auth-form__container {
        width: 113.1rem;
    }

    .auth-form__box-field {
        width: 36.8rem;
    }    
}
