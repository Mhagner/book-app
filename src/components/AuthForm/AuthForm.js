import React from 'react'
import { ErrorMessage, Field, Formik, Form } from 'formik'
import * as yup from 'yup'

import './AuthForm.css'
import Logo from '../Logo/Logo'


const AuthForm = ({ handleSubmit, errors }) => {
   
    const validations = yup.object().shape({
        email: yup.string().email().required(),
        password: yup.string().min(8).required()
    })

    return (
        <div data-testid="auth-form" className="auth-form__container">
            <Logo />
            <Formik
                initialValues={{}}
                onSubmit={handleSubmit}
                validationSchema={validations}
            >
                <Form className="auth-form__form">
                    <div className="auth-form__box-field">
                        <label>Email</label>
                        <Field
                            data-testid="auth-input-email"
                            name="email"
                            className="auth-form__field"
                        />
                        <ErrorMessage
                            data-testid="auth-msg-error-email"
                            component="span"
                            name="email"
                            className="auth-form__error"
                        />
                    </div>
                    <div className="auth-form__box-field">
                        <label>Senha</label>
                        <Field
                            data-testid="auth-input-password"
                            name="password"
                            className="auth-form__field"
                            type="password"
                        />
                        <ErrorMessage
                            data-testid="auth-msg-error-password"
                            component="span"
                            name="password"
                            className="auth-form__error"
                        />
                        <button
                            data-testid="auth-button-submit"
                            className="auth-form__button"
                            type="submit"
                        >
                            Entrar
                        </button>
                        {errors &&
                            <span className="error-backend">{errors}</span>
                        }
                    </div>
                </Form>
            </Formik>
        </div>
    )
}

export default AuthForm