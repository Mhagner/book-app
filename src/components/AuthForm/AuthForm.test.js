import { render } from '@testing-library/react'

import AuthForm from './AuthForm'

describe('AuthForm component', () => {
    const renderComponent = () => {
        return render(<AuthForm />)
    }

    it('Componente should defined', () => {
        const { getByTestId } = renderComponent()

        const element = getByTestId('auth-form')

        expect(element).toBeDefined()
    })

    it('Componente should render', () => {

        const { getByTestId } = renderComponent()

        const element = getByTestId('auth-form')

        expect(element).toMatchSnapshot()
    })

})