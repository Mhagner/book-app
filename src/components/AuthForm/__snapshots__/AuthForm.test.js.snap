// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm component Componente should render 1`] = `
<div
  class="auth-form__container"
  data-testid="auth-form"
>
  <div
    class="logo__container"
    data-testid="logo"
  >
    <img
      alt="ioasys"
      class="logo__image"
      src="logo.png"
    />
    <span
      class="logo__text  logo__text--light"
    >
      Books
    </span>
  </div>
  <form
    action="#"
    class="auth-form__form"
  >
    <div
      class="auth-form__box-field"
    >
      <label>
        Email
      </label>
      <input
        class="auth-form__field"
        data-testid="auth-input-email"
        name="email"
        value=""
      />
    </div>
    <div
      class="auth-form__box-field"
    >
      <label>
        Senha
      </label>
      <input
        class="auth-form__field"
        data-testid="auth-input-password"
        name="password"
        type="password"
        value=""
      />
      <button
        class="auth-form__button"
        data-testid="auth-button-submit"
        type="submit"
      >
        Entrar
      </button>
    </div>
  </form>
</div>
`;
